pkg/pulumiyaml/run_test.go

```diff
@@ -1640,39 +1638,6 @@ variables:
 	})
 }
 
-// TestReadFileForbidsPathTraversal ensures that we forbid certain malicious path behaviors which
-// allow escaping the project directory in static YAML.
-//
-// The example program uses a non-constant path which escapes the project directory.
-//
-// Non-constant paths which read from the project dir are considered safe, likely as uses of
-// ${pulumi.cwd}, see above. Constant, absolute path are also permitted, sometimes necessary to use
-// a secret or token.
-func TestReadFileForbidsPathTraversal(t *testing.T) {
-	t.Parallel()
-
-	text := `
-name: test-readfile
-runtime: yaml
-outputs:
-  readme:
-    fn::readFile: ${pulumi.cwd}/../../go.mod # imagine this is /etc/shadow, /var/run/secrets/tokens, etc.
```

Comment:
This made me pause for a second, as it's security related.  I don't think it's in our security model to guard against users doing anything in their programs in general though (that would be really hard to do in other supported languages anyway).  The user running the program needs to be the one to inspect and trust it, so I think removing this restriction is fine.

Reply:
That was my thinking